### PR TITLE
[training] Add Cyclical Tracker

### LINKED ISSUE #37: Dynamic learning parameter - time period
## Description
Dynamically modified learning parameter over time. We want learning parameter to change dynamically when network is learning:
- constant learning parameter at the start then **cycle** mode. We want to start with constant learning parameter, then after N epoch start **cycle** - like saw, from min. to max. definition of learning parameter (up and down)
- it should be configurable - we want to start with static learning parameter and then **cycle**, we want to start with **cycle** at the beginning,.. 
- any combination with constant and **cycle**, for example 0-10 epoch constant learning parameter, then **cycle**(whatever type) (11-20), then again constant parameter (21-30),..

**Cycle** (learning parameter alternation) mode should have multiple types, with configurable time interval and min., max. learning parameter value, for example:
- like saw wave (up and down /\ /\ /\ )
- like first half of saw wave (just up / / / )
- like second half of saw wave (just down \ \ \ )
- we can go any combination (/\ - / / - \ \ )

### Will this change the current api? How?
Yes, implement learning parameter which dynamically changes over iteration/epoch

### Who will benefit from this feature?
Everybody, this will enable network to learn more quickly.

## References
Here is dynamic learning parameter change implementation, kind of saw wave over time:
https://github.com/eclipse/deeplearning4j/blob/b5f0ec072f3fd0da566e32f82c0e43ca36553f39/nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/schedule/CycleSchedule.java
We want to extend this implementation - see above.

### PR DESCRIPTION
**Pull request by dandansamax**
## Description ##

Add Cyclical Tracker according to cyclical learning rate policy, as detailed in the paper [Cyclical Learning Rates for Training Neural Networks](https://arxiv.org/abs/1506.01186).

close #37.

### CODE REVIEW

File: api/src/main/java/ai/djl/training/tracker/CyclicalTracker.java [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    private int stepSizeDown;
+    private int totalSize;
+    private float stepRatio;
+    private ScaleFunction scaleFunction;
+    private boolean scaleModeCycle;
+
+    /**
+     * Creates a new instance of {@code CosineTracker}.
+     *
+     * @param builder the builder to create a new instance of {@code CosineTracker}
```
Comment (zachgk) [inline api/src/main/java/ai/djl/training/tracker/CyclicalTracker.java:37]:
```suggestion
     * Creates a new instance of {@code CyclicalTracker}.
     *
     * @param builder the builder to create a new instance of {@code CyclicalTracker}
```

**Review by zachgk — APPROVED** [review]

**Review by frankfliu — APPROVED** [review]
